### DISCUSSION TITLE
How to get the rpm package to install cloudberrydb ?

### DISCUSSION THREAD

**Discussion by jose-l-goncalves-alb** [apache/cloudberry#463] [Q&A]
Hello,

We have been used greenplumdb. We used the rpm package (open-source-greenplum-db-7.1.0-el8-x86_64.rpm) that were available at greenplum-db github to download.

After the end of greenplumdb, we are going to evaluate cludberrydb.
At ["Step 2. Install the RPM package"@"Deploy Cloudberry Database Manually Using RPM Package"](https://cloudberrydb.org/docs/cbdb-op-deploy-guide#step-2-install-the-rpm-package) page we see:
```
After the preparation, it is time to install Cloudberry Database.
You need to download the corresponding RPM package from Cloudberry Database Releases, and then install the database on each node using the installation package.

Download the RPM package to the home directory of gpadmin.

wget -P /home/gpadmin <download address>
```
What is the "download address" to get the rpm ?
I cannot find any rpm file at https://github.com/cloudberrydb/cloudberrydb/releases

Or how can I build the rpm to install cloudberrybd at all nodes of database cluster (coordinator & segment nodes) ?

Thank you for your help

**Comment by tuhaihe** [comment]
Hi @jose-l-goncalves-alb thanks for choosing Cloudberry as your option. The RPM packagege is still not in public, but there is one pr #459 to build weekly, which will be merged in the following days.

For now, you can choose to build from source code.

**Reply by zlokomatic** [reply]
What i did was to use https://github.com/cloudberrydb/bootcamp/blob/main/000-cbdb-sandbox/cbdb_Dockerfile for a starting point to build from source so you know what libraries are needed etc.

**Reply by edespino** [reply]
@zlokomatic & @jose-l-goncalves-alb - If you need any assistance building or general product questions, you can join the Slack instance to look for assistance: https://cloudberrydb.org/community/slack

**Comment by francksidi-edb** [comment]
Hi any update for getting a RPM for evaluation ?

**Reply by tuhaihe** [reply]
Hi, the newer version 1.5.4 will be released in the next week. Plan to release the rpm package for the test.

**Reply by francksidi-edb** [reply]
Thanks a lot, will you include support also for pgvector extension and other AI extensions ? what's about PL-Rust support 
Well done. I am a former Greenplum person, worked for 13 years and I am impressed by how fast you went to modernize this nice solution.

**Reply by tuhaihe** [reply]
Hi @francksidi-edb, thanks for your reply. We can support if you need them, but with limited resources, the work may not be done in a short time. Hope you understand this. If you want to join to maintain, we would like to have you.

**Comment by francksidi-edb** [comment]
Also what is the solution to stream data inside Cloudberry and how can I stream data including unstructured data that I will store using the directory table feature ?

**Reply by tuhaihe** [reply]
Hi, here is one doc on the directory table feature: https://cloudberrydb.org/docs/advanced-analytics/directory-tables. You can learn more.

**Reply by francksidi-edb** [reply]
Hi I already read the doc. My question was how to stream data to Cloudberry and put structured data inside a regular table and push the unstructured data inside a directory table. does it make sense ?

**Reply by jianlirong** [reply]
We plan to provide two kinds of streaming ingestion solution. One is based on the kafka connector framework(just like the snowpipe from Snowflake), the other is the kafka_fdw extension. Both of these two solutions assume users have already streamed their data into a kafka topic. The difference between these two solutions is that: the former can support CDC data streams (not just appending data, also deleting and updating), and it depends on the kafka connector framework; the latter only support appending data, and has no external dependency, each segment consuming data directly from the kafka topic.

We plan to open source both of these solutions in the near future. Thanks.

**Reply by francksidi-edb** [reply]
Thanks a lot when will it be ready or planned ?

**Reply by jianlirong** [reply]
All the code of these two solutions has been ready in the internal code repository, and deployed in the production environment of some commercial customers. We just need to do some code refactoring and cleaning up before open source. Hopefully we can make it before 2024 Q4.
